Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml

```diff
@@ -18449,4 +18426,224 @@
+        <EffectArray Operation="Divide" Reference="Abil,AP_SummonZergMercs,InfoArray[Train3].Cooldown.TimeStart" Value="2"/>
+        <EffectArray Operation="Divide" Reference="Abil,AP_SummonZergMercs,InfoArray[Train4].Cooldown.TimeStart" Value="2"/>
+        <EffectArray Operation="Divide" Reference="Abil,AP_SummonZergMercs,InfoArray[Train5].Cooldown.TimeStart" Value="2"/>
+        <EffectArray Operation="Divide" Reference="Abil,AP_SummonZergMercs,InfoArray[Train6].Cooldown.TimeStart" Value="2"/>
+        <EffectArray Operation="Divide" Reference="Abil,AP_SummonZergMercs,InfoArray[Train7].Cooldown.TimeStart" Value="2"/>
+        <EffectArray Operation="Divide" Reference="Abil,AP_SummonZergMercs,InfoArray[Train8].Cooldown.TimeStart" Value="2"/>
+        <EffectArray Operation="Divide" Reference="Abil,AP_SummonZergMercs,InfoArray[Train9].Cooldown.TimeStart" Value="2"/>
+        <EffectArray Operation="Divide" Reference="Abil,AP_SummonZergMercs,InfoArray[Train10].Cooldown.TimeStart" Value="2"/>
+    </CUpgrade>
+    <CUpgrade id="AP_EvolutionaryLeapLevel2">
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train1].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train2].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train3].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train4].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train5].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train6].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train7].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train8].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train9].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train10].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train11].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train12].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train13].Cooldown.TimeStart" Value="0"/>
+        <EffectArray Operation="Set" Reference="Abil,AP_SummonMercenaries,InfoArray[Train14].Cooldown.TimeStart" Value="0"/>
+    </CUpgrade>
+    <CUpgrade id="AP_FastDeliveryLevel1"/>
+    <CUpgrade id="AP_SelfSufficient">

```

Comment:
Some of the units listed there have an upgrade that cuts their supply (Hydra, Tank, Medic, Ultra) as both `Add` and `Set` operations are used there. Therefore in the unlock trigger of those upgrades check if the player has `AP_SelfSufficient` (as he can unlock `AP_SelfSufficient` first, then the unit-specific upgrade) and reset the value to 0 using a trigger in order to prevent those units being supply providers in some cases